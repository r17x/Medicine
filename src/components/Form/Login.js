import React, { PureComponent } from 'react'
import Form from 'utils/Form'
import Input from 'antd/lib/input'
// import Upload from 'antd/lib/upload'
// import Icon from 'antd/lib/icon'

// const {Dragger} = Upload

export default class FormLogin extends PureComponent {
  constructor(props) {
    super(props)
    this.state = {
      fields: [
        {
          id: 'username',
          options: {
            rules: [
              {
                required: true,
              },
            ],
          },
          element: <Input placeholder="Username / Email" />,
        },
        {
          id: 'password',
          options: {
            rules: [
              {
                required: true,
              },
            ],
          },
          element: <Input type="password" placeholder="Password..." />,
        },
      ],
    }
  }

  render() {
    return (
      <Form save={this.props.save} title="Login" fields={this.state.fields} />
    )
  }
}
