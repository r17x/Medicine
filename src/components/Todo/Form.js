import React, { PureComponent } from 'react'
import { func } from 'prop-types'
import Form from 'utils/Form'
import Input from 'antd/lib/input'

export default class FormTodo extends PureComponent {
  static propTypes = {
    save: func,
  }

  state = {
    fields: [
      {
        id: 'title',
        options: {
          rules: [
            {
              required: true,
            },
          ],
        },
        element: <Input placeholder="Create your Todo..." />,
      },
    ],
  }

  render() {
    return (
      <Form
        save={this.props.save}
        title="Todo"
        fields={this.state.fields}
        hideCancel
      />
    )
  }
}
