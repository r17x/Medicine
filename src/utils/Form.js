import React, { PureComponent } from 'react'
import { string, array, object, func } from 'prop-types'
import Form from 'antd/lib/form'
import Button from 'antd/lib/button'

const { Item } = Form
export default Form.create()(
  class FormExport extends PureComponent {
    static propTypes = {
      layout: string.isRequired,
      fields: array.isRequired,
      form: object,
      title: string,
      save: func,
      data: object,
      errors: object,
    }

    static defaultProps = {
      layout: 'horizontal',
      fields: [],
      title: '',
      hideCancel: false,
      hideSubmit: false,
      submitText: 'Submit',
    }

    constructor(props) {
      super(props)
      this.renderFields = this.renderFields.bind(this)
      this.handleSubmit = this.handleSubmit.bind(this)
    }

    syncDataWithFields = () => {
      const values = this.props.form.getFieldsValue()
      const { data, errors } = this.props
      Object.keys(values).forEach(key => {
        this.props.form.setFields({
          [key]: {
            value: data[key] || values[key],
            errors:
              errors === Object(errors) && key in errors
                ? [new Error(errors[key])]
                : undefined,
          },
        })
      })
    }

    renderFields({ id, options, element }) {
      return (
        <Item key={id}>
          {this.props.form.getFieldDecorator(id, options)(element)}
        </Item>
      )
    }

    validateFields = (err, values) => {
      if (err) return
      console.log(values)
      this.props.save(values)
    }

    handleSubmit(e) {
      e.preventDefault()
      this.props.form.validateFields(this.validateFields)
    }

    handleCancel = e => {
      e.preventDefault()
      this.props.form.resetFields()
    }

    render() {
      return (
        <Form
          style={{ width: '100%' }}
          layout={this.props.layout}
          onSubmit={this.handleSubmit}
        >
          <h2> {this.props.title} </h2>
          {this.props.fields.map(this.renderFields)}
          <Item>
            {!this.props.hideSubmit && (
              <Button type="primary" htmlType="submit">
                {this.props.submitText}
              </Button>
            )}
            {'   '}
            {!this.props.hideCancel && (
              <Button onClick={this.handleCancel}>Cancel</Button>
            )}
          </Item>
        </Form>
      )
    }
  },
)
