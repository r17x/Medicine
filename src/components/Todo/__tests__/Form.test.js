import React from 'react'
import Form from '../Form'

describe('Form', () => {
  it('should render correctly in "debug" mode', () => {
    const component = shallow(<Form debug />)
    expect(component).toMatchSnapshot()
    component.unmount()
  })

  it('Form mount with no error', () => {
    const component = mount(<Form />)
    component.unmount()
  })
})
