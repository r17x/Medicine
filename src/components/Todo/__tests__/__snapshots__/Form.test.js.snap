// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Form should render correctly in "debug" mode 1`] = `
<Form(FormExport)
  fields={
    Array [
      Object {
        "element": <Input
          disabled={false}
          placeholder="Create your Todo..."
          prefixCls="ant-input"
          type="text"
        />,
        "id": "title",
        "options": Object {
          "rules": Array [
            Object {
              "required": true,
            },
          ],
        },
      },
    ]
  }
  hideCancel={true}
  title="Todo"
/>
`;
